torchtext/experimental/vectors.py

```diff
@@ -144,7 +144,7 @@ def GloVe(name="840B", dim=300, unk_tensor=None, root=".data", validate_file=Tru
     return vectors_obj
 
 
-def vectors_from_file_object(file_like_object, delimiter=",", unk_tensor=None, num_cpus=10):
+def load_vectors_from_file_object(file_like_object, delimiter=",", unk_tensor=None, num_cpus=10):
```

Comment:
This actually assumes the file_object comes from a path via open. We should just restrict this to accept paths only. We should then also add support for [pathlib](https://docs.python.org/3/library/pathlib.html) while we're at it.